csrc/scheduler/mma_utils.cpp

```diff
@@ -1812,15 +1819,47 @@ MmaOp* MatmulPattern::translateToMmaOp() {
     NVF_ERROR(
         A->nDims() > 1 && B->nDims() > 1,
         "Cannot translate LinearOp with 1D input");
-    std::vector<bool> bcast_dim((size_t)A->nDims() + 1, false);
-    bcast_dim[bcast_dim.size() - 2] = true; // N
-    A = broadcast(A, bcast_dim);
+    NVF_ERROR(
+        B->nDims() == 2, "Cannot translate LinearOp without 2D weight tensor");
+    if (avoid_intermediates) {
+      MmaOp::AxisMapping axis_mapping;
+      size_t out_dim = (size_t)A->nDims() + 1;
```

Comment:
nitpick: It seems like the back and forth cast from `size_t` and `int64_t` is unnecessary.
```suggestion
      int64_t out_dim = A->nDims() + 1L;
      axis_mapping.a_axes.reserve(out_dim);
      for (int64_t d : c10::irange(out_dim - 2L)) {
        axis_mapping.a_axes.push_back(d);
      }
```